src/modules/ekf2/EKF/aid_sources/external_vision/ev_vel_control.cpp

```diff
@@ -111,48 +113,56 @@ void Ekf::controlEvVelFusion(const extVisionSample &ev_sample, const bool common
 	// increase minimum variance if GPS active (position reference)
 	if (_control_status.flags.gps) {
 		for (int i = 0; i < 2; i++) {
-			vel_cov(i, i) = math::max(vel_cov(i, i), sq(_params.gps_vel_noise));
+			measurement_var(i) = math::max(measurement_var(i), sq(_params.gps_vel_noise));
 		}
 	}
 
 #endif // CONFIG_EKF2_GNSS
 
-	const Vector3f measurement{vel};
-
-	const Vector3f measurement_var{
-		math::max(vel_cov(0, 0), sq(_params.ev_vel_noise), sq(0.01f)),
-		math::max(vel_cov(1, 1), sq(_params.ev_vel_noise), sq(0.01f)),
-		math::max(vel_cov(2, 2), sq(_params.ev_vel_noise), sq(0.01f))
+	measurement_var = Vector3f{
+		math::max(measurement_var(0), minimum_variance),
+		math::max(measurement_var(1), minimum_variance),
+		math::max(measurement_var(2), minimum_variance)
 	};
+	continuing_conditions_passing &= measurement.isAllFinite() && measurement_var.isAllFinite();
+
+	if (ev_sample.vel_frame == VelocityFrame::BODY_FRAME_FRD) {
+		const Vector3f measurement_var_ekf_frame = rotateVarianceToEkf(measurement_var);
+		const Vector3f measurement_ekf_frame = _R_to_earth * measurement;
+		const uint64_t t = aid_src.timestamp_sample;
+		updateAidSourceStatus(aid_src,
```

Comment:
We shouldn't be calling updateAidSourceStatus() multiple times now that it has filtering internally.

Reply:
It only gets called once. Although I'm overwriting values afterwards which is not optimal. I'm planning to do an overall cleanup, because I think we need to split these two methods a bit more. I'm currently calling it in the beginning to have a test ratio for the starting condition, where I'm using the transformed measurement...